### PR TITLE
#3 Bugfix: Safer JSON parsing of cookieData, and validation of the parsed cookie

### PR DESCRIPTION
**Pull request by david-ma**
It works on my machine, I hope this makes sense for you :)

### CODE REVIEW

File: app/page.tsx [hunk 1]
```diff
@@ -58,7 +58,25 @@ function Cookies() {
...
+
+        // Check cookie is a valid Cookie
+        if (
+          typeof cookie.id !== 'number' ||
+          typeof cookie.imgSrc !== 'string' ||
+          typeof cookie.centerX !== 'number' ||
+          typeof cookie.centerY !== 'number'
+        ) {
+          throw new Error('Invalid Cookie');
+        } else {
```
Comment (leah) [inline app/page.tsx]: I don't think `else` is needed here since it throws.

File: app/page.tsx [hunk 2]
```diff
@@ -58,7 +58,25 @@ function Cookies() {
...
+          initialCookies.push(cookie)
+        }
+      } catch (e) {
+        console.error('Error parsing cookie data:')
+        console.log(cookieData)
+        console.log(e)
```
Comment (leah) [inline app/page.tsx]: Could you remove the console.logs?

**Review by leah — CHANGES_REQUESTED** [review]
Minor comments - are you okay updating?

File: app/page.tsx (hunk 1 above)
Comment (david-ma) [reply app/page.tsx]: Ah yes, that isn't needed.

File: app/page.tsx (hunk 2 above)
Comment (david-ma) [reply app/page.tsx]: Is console.error ok? It only prints if an error happens.

Comment (david-ma) [reply app/page.tsx]: I've removed the console.logs

**Review by leah — APPROVED** [review]

**Comment by leah** [comment]
Looks great, thanks!

**Pull request merged**